runtime/src/main/java/org/corfudb/runtime/LogReplicationUtils.java

```diff
@@ -86,6 +86,50 @@ public static void subscribe(@Nonnull LogReplicationListener clientListener, @No
         log.info("Client subscription at timestamp {} successful.", subscriptionTimestamp);
     }
 
+    public static void subscribeRqListener(@Nonnull LogReplicationRoutingQueueListener clientListener, @Nonnull String namespace,
+                                           int bufferSize, CorfuStore corfuStore) {
+
+        long subscriptionTimestamp = getRoutingQueueSubscriptionTimestamp(corfuStore, namespace, clientListener);
+        // Open the routing queue from corfu store (Disk backed mode)
+        // since the subscribe API will run in policy jvm
+        // TODO: Get client name as input params. For now, hard coding the stream tag.
+        if (checkIfRoutingQueueExists(corfuStore, namespace, REPLICATED_QUEUE_TAG_PREFIX)) {
+            // Table registry contains the routing queue already.
+            corfuStore.getRuntime().getTableRegistry().getStreamingManager().subscribeLogReplicationRoutingQueueListener(
```

Comment:
subscription must be done at a point where getDataConsistent() == true.  Did not find it being checked...

Reply:
getRoutingQueueSubscriptionTimestamp makes use of getDataConsistent() call and marks sync inprogress flags, so listener won't process and returns. Let me know if you're referring to something else.